java-frontend/src/test/java/org/sonar/java/model/JSymbolTest.java

```diff
@@ -660,6 +662,101 @@ interface F extends java.util.function.Consumer<String> {}
     assertNotEquals(methodSymbol1, methodSymbol2);
   }
 
+  @Test
+  void testEquality_oneLambda() {
+    CompilationUnitTree cu = test("""
+      class C {
+        java.util.function.Consumer<String> k = s -> {};
+        void j(String s) {};
```

Comment:
For these 2 to be equivalent, `j` would need to be static
```suggestion
        java.util.function.Consumer<String> k = s -> {};
        static void j(String s) {};
```